### PR TITLE
fix: can't join an encrypted community

### LINKED ISSUE #14299: During the join / edit shared addresses flow, if requirements check backend fails, UI doesn't act accordingly
### Description
When selecting / unselecting account that contains tokens related to join permissions, the requirements check works the first time, but not the second one, triggering an error in log but the UI doesn't evaluates this result so that the user is able to continue with the join flow. In the following video the failure happens at the end but you can see from the very begining, that the initial behaviour is correct, then after more selections / unselections the flow becomes incorrect.

https://github.com/status-im/status-desktop/assets/97019400/10fea224-f4c5-4bf6-8872-c43fbe1ecb1e

**Log**:
```
ERR 2024-04-04 17:47:01.782+02:00 error doing rpc request                    
topics="rpc" tid=2251947 file=core.nim:40 methodName=wakuext_checkAllCommunityChannelsPermissions exception="\nstatus-go error [methodName:wakuext_checkAllCommunityChannelsPermissions, code:-32000, message:fallback failed with 'unsuccessful request: 403 Forbidden'. run error was 'hystrix: timeout' ]\n"
```

**Posible solutions:**
- Disable the dialog button while the app is in requirements checking state (ask _design team_ about this disable state).
- Backend to emit a signal on failure and UI to listen to it to act accordingly.

### LINKED ISSUE #14473: Can't join an encrypted community
# Bug Report

## Description

Can't  joining an encrypted community, share addresses button is disabled if even conditions are matching.

Screenshot 2024-04-18 at 15 58 55

## Steps to reproduce

1. Create a community
2. Add a permission, for example to view & post in a channel 
Screenshot 2024-04-18 at 16 41 45

3. Try to join without satisfied channel permissions
Screenshot 2024-04-18 at 15 07 28

4. Try to join with satisfied channel permissions
Screenshot 2024-04-18 at 15 58 55

5. Remove permission
6. Try to join again
Screenshot 2024-04-18 at 16 26 13

#### Expected behavior

A user is able to share addresses and join on on steps 3, 4 and 6

#### Actual behavior

A user is not able to share addresses and join on on steps 3, 4 and 6

### Additional Information

- Status desktop version: current. master
- Network: test net

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

- fix the corner case and allow the user to join a community without an explicitly stated "Member" permission
- enable/disable the Share/Join/Save buttons when the permission to join check is ongoing, or when the permission check failed
- display tooltips over the disabled buttons explaining why it's disabled
- always display the eligibility button floating on top of the (scrollable) contents

Fixes #14473
Fixes #14299

### Affected areas

CommunityMembershipSetupDialog

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Main join dialog:
Snímek obrazovky z 2024-04-25 12-24-03

Check over:
Snímek obrazovky z 2024-04-25 12-24-11

Editting addresses:
Snímek obrazovky z 2024-04-25 12-22-14

Snímek obrazovky z 2024-04-25 12-24-20

### CODE REVIEW

File: ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml [hunk 1]
```diff
@@ -101,11 +102,27 @@ Control {
...
+
         readonly property var saveButton: StatusButton {
-            enabled: d.dirty
-            type: d.lostCommunityPermission || d.lostChannelPermissions ? StatusBaseButton.Type.Danger : StatusBaseButton.Type.Normal
             visible: root.isEditMode
+            interactive: d.dirty && !root.requirementsCheckPending && root.joinPermissionsCheckSuccessful
+            loading: root.requirementsCheckPending
+            type: d.lostCommunityPermission || d.lostChannelPermissions ? StatusBaseButton.Type.Danger : StatusBaseButton.Type.Normal
+            tooltip.text: {
+                if (interactive)
```
Comment (MishkaRogachev) [inline ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml:121]: is tooltip available if !interactive

**Review by MishkaRogachev — APPROVED** [review]
Looks good!

File: ui/StatusQ/src/permissionutilsinternal.cpp [hunk 2]
```diff
@@ -161,35 +161,37 @@ int /*PermissionTypes::Type*/ PermissionUtilsInternal::isEligibleToJoinAs(QAbstr
...
+        return type == PermissionTypes::Type::Member;
+    };
+
     const auto permissionsCount = permissionsModel->rowCount();
     for (int i = 0; i < permissionsCount; i++) {
         const auto permissionType = static_cast<PermissionTypes::Type>(permissionsModel->data(permissionsModel->index(i, 0), permissionTypeRole).toInt());
         if (isJoinTypePermission(permissionType)) {
-            hasAnyJoinPermission = true;
+            if (isMemberPermission(permissionType))
+                hasMemberPermission = true;
```
Comment (alexjba) [inline ui/StatusQ/src/permissionutilsinternal.cpp:180]: `isMemberPermission` is used in a single place. Could be simplified with this

```suggestion
                hasMemberPermission = permissionType == PermissionTypes::Type::Member;
```

**Review by alexjba — APPROVED** [review]
LGTM!
